### PR TITLE
restaurant chains update - return json format data

### PR DESCRIPTION
**Pull request by sylcym**

### CODE REVIEW

File: controllers/restaurantsChainController.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { httpStatus } from '../utils/httpStatusCode.js';
+import { restaurantsChainService } from '../services/restaurantsChainService.js';
+
+const getChains = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.getChains(req.query);
+
+    if (chain.length === 0) {
+      return res.status(httpStatus.NO_CONTENT).json({
```
Comment (radekwojpl2) [inline controllers/restaurantsChainController.js]: think about it, does it make sense to check length here? if we do not have records in database maybe ww should return empty array not not 204 no content, read about this status code here 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

File: controllers/restaurantsChainController.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    });
+  }
+};
+
+const getChainById = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.getChainById(req.params);
+    if (!chain) {
+      return res.status(httpStatus.NO_CONTENT).json({
+        message: 'ID not found.',
```
Comment (radekwojpl2) [inline controllers/restaurantsChainController.js:41]: I would return 404, id is not found in db and that is an error

File: controllers/restaurantsChainController.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      error: 'Server error',
+    });
+  }
+};
+
+const deleteChainById = async function (req, res) {
+  try {
+    await restaurantsChainService.deleteChainById(req.params);
+
+    return res.status(httpStatus.NO_CONTENT).json({
```
Comment (radekwojpl2) [inline controllers/restaurantsChainController.js]: same here, i would return 404 not 204

File: controllers/restaurantsChainController.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  } catch (error) {
+    return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+      success: false,
+      error: 'Server error',
+    });
+  }
+};
+
+export const restaurantsChainController = {
+  getChains,
```
Comment (radekwojpl2) [inline controllers/restaurantsChainController.js]: getAllChains vs getChains ? which one is better

File: services/restaurantsChainService.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      chainToUpdate,
+    );
+    return updatedChain.save();
+  } catch (error) {
+    throw Error('Error while editing chain');
+  }
+};
+
+export const restaurantsChainService = {
+  getChains,
```
Comment (radekwojpl2) [inline services/restaurantsChainService.js]: same here, consider different name for this function

File: utils/restaurantChainsRouterValidation.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+export const validateRestaurantChainId = (req, res, next) => {
+  const schema = Joi.object({
+    id: Joi.string().hex().length(24),
+  });
+  const { id } = req.params;
+  const { error } = schema.validate({ id });
+
+  if (error) {
+    const message = error.details[0].message;
+    return res.status(httpStatus.UNPROCESSABLE_ENTITY).json({ error: message });
```
Comment (radekwojpl2) [inline utils/restaurantChainsRouterValidation.js:28]: i would return 404, i think 422 is for content type not for validation errors 
read about it here https://datatracker.ietf.org/doc/html/rfc4918#section-11.2

File: controllers/restaurantsChainController.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import { httpStatus } from '../utils/httpStatusCode.js';
+import { restaurantsChainService } from '../services/restaurantsChainService.js';
+
+const getChains = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.getChains(req.query);
```
Comment (radekwojpl2) [inline controllers/restaurantsChainController.js]: your controller should only be responsible for validation, authorisation/authentication not for logic, 
you can consider moving this logic to service layer

File: controllers/restaurantsChainController.js (hunk 1 above)
Comment (sylcym) [reply controllers/restaurantsChainController.js]: done

File: controllers/restaurantsChainController.js (hunk 2 above)
Comment (sylcym) [reply controllers/restaurantsChainController.js:41]: done

File: controllers/restaurantsChainController.js (hunk 3 above)
Comment (sylcym) [reply controllers/restaurantsChainController.js]: done

File: controllers/restaurantsChainController.js (hunk 4 above)
Comment (sylcym) [reply controllers/restaurantsChainController.js]: done

File: utils/restaurantChainsRouterValidation.js (hunk 6 above)
Comment (sylcym) [reply utils/restaurantChainsRouterValidation.js:28]: done

File: controllers/restaurantsChainController.js (hunk 7 above)
Comment (sylcym) [reply controllers/restaurantsChainController.js]: according to node.js tutorial, controllers handles all the logic behind validating request parameters, query, sending responses with correct codes, so I think this logic is ok (also I took a look at Hania code and she handled it the same way)

File: services/restaurantsChainService.js (hunk 5 above)
Comment (sylcym) [reply services/restaurantsChainService.js]: done

**Pull request merged**